packages/amplify-graphql-transformer-core/src/transformer-context/resolver.ts

```diff
@@ -282,7 +289,7 @@ export class TransformerResolver implements TransformerResolverProvider {
         case 'AMAZON_DYNAMODB':
           if (this.datasource.ds.dynamoDbConfig && !isResolvableObject(this.datasource.ds.dynamoDbConfig)) {
             const tableName = this.datasource.ds.dynamoDbConfig?.tableName;
-            dataSource = `$util.qr($ctx.stash.put("tableName", "${tableName}"))`;
+            dataSource = stashStatement('tableName', `"${tableName}"`);
```

Comment:
It seems brittle to make the caller responsible for remembering to include double quotes, and I'm worried about the error that will happen when we revisit this in a few months, have forgotten the details, and do a copypasta error leading to a hard-to-find bug. Can we adjust the stashStatement API to make it clear how to pass a value:

```typescript
stashStatement('tableName', { quote: tableName })
...
stashStatement('conditions', { literal: '[]' })
```

or

```ts
stashString('tableName', tableName)
...
stashExpression('tableName', '[]')
```

or some other clearer way of passing values?

(I know these kinds of sharp syntax edges are all over the codebase--believe me I know!--but I'd love for us to be the change we want to see in the code.)

Reply:
Yes, definitely a sharp edge. I figured it was no worse than before, but that's not following the boy scout rule very well. I like the `stashString` / `stashExpression` suggestion.

Addressed in [113d921](https://github.com/aws-amplify/amplify-category-api/pull/2812/commits/113d921c23ec8d9262af4a05dca7705b881dcb82).